src/transformers/pipelines/image_classification.py

```diff
@@ -89,6 +89,13 @@ def __call__(self, images: Union[str, List[str], "Image.Image", List["Image.Imag
     def preprocess(self, image):
         image = load_image(image)
         model_inputs = self.feature_extractor(images=image, return_tensors=self.framework)
+
+        if type(self.model).__name__ == "ImageGPTForImageClassification":
+            # Temporary workaround
+            # Check that the model can still do a forward pass successfully (every parameter should be resized)
+            # Input ids should be clamped to the maximum size of the vocabulary
+            model_inputs["pixel_values"].clamp_(max=self.model.config.vocab_size - 15 - 1)
```

Comment:
`ImageGPTModelTester` has `vocab_size=99`, but `ImageGPTFeatureExtractor` doesn't use `vocab_size`, and will produce values between 0 and 512.

This operation is copied from `ImageGPTModelTest`.

Reply:
I didn't look the details in `ImageGPT`. It seems to me it is special: unlike other vision models' feature extractors which output pixel values, `ImageGPTFeatureExtractor` outputs something called `color clusters` which should be treated like tokens as in NLP models.

In fact, the following message is in `ImageGPTModel`

```python
        if "pixel_values" in kwargs:
            warnings.warn(
                "The `pixel_values` argument is deprecated and will be removed in a future version, use `input_ids` instead.",
                FutureWarning,
            )
```

Reply:
Probably we can move this logic to be inside `ImageGPTFeatureExtractor` itself. But this is not done for any vision model, and @NielsRogge 's opinion is important for this change.

Reply:
I see.